File: ruby/download_firefox.rb
#!/usr/bin/env ruby

require 'rubygems'
require 'selenium-webdriver'

profile = Selenium::WebDriver::Firefox::Profile.new
profile['browser.download.folderList'] = 0
profile['browser.download.manager.showWhenStarting'] = false
profile['browser.download.manager.focusWhenStarting'] = false
profile['browser.download.useDownloadDir'] = true
profile['browser.helperApps.alwaysAsk.force'] = false
profile['browser.download.manager.alertOnEXEOpen'] = false
profile['browser.download.manager.closeWhenDone'] = true
profile['browser.download.manager.showAlertOnComplete'] = false
profile['browser.download.manager.useWindow'] = false
profile['browser.helperApps.neverAsk.saveToDisk'] = "application/java-archive,application/x-compressed"

capabilities = Selenium::WebDriver::Remote::Capabilities.firefox(:firefox_profile => profile)
capabilities['browserstack.debug'] = "true"

driver = Selenium::WebDriver.for(:remote,
  :url => "http://BS_USERNAME:BS_AUTHKEY@hub.browserstack.com/wd/hub", 
  :desired_capabilities => capabilities)

driver.navigate.to "http://www.browserstack.com/BrowserStackTunnel.jar"
sleep(5)
driver.quit

File: perl/download_firefox.pl
#!/usr/bin/perl

use Selenium::Remote::Driver; 

my $login = "BS_USERNAME";
my $key = "BS_AUTHKEY";

my $host = "$login:$key\@hub.browserstack.com";
my $browser = "firefox";

my $driver = new Selenium::Remote::Driver(
  'remote_server_addr' => $host,
  'port' => '80',
  'browser_name' => $browser,
  'extra_capabilities' => {
    'firefox_profile' => {
      'browser.download.folderList' => 0,
      'browser.download.manager.showWhenStarting' => 0,
      'browser.download.manager.focusWhenStarting' => 0,
      'browser.download.useDownloadDir' => 1,
      'browser.helperApps.alwaysAsk.force' => 0,
      'browser.download.manager.alertOnEXEOpen' => 0,
      'browser.download.manager.closeWhenDone' => 1,
      'browser.download.manager.showAlertOnComplete' => 0,
      'browser.download.manager.useWindow' => 0,
      'browser.helperApps.neverAsk.saveToDisk' => "application/java-archive,application/x-compressed"
    }
  }
);

$driver->get('http://www.browserstack.com/BrowserStackTunnel.jar');

sleep(5);

$driver->quit();
